ecs-agent/tmds/handlers/fault/v1/types/types.go

```diff
@@ -13,8 +13,59 @@
 
 package types
 
+import "fmt"
+
 const (
-	BlackHolePortFaultType = "network-blackhole-port"
-	LatencyFaultType       = "network-latency"
-	PacketLossFaultType    = "network-packet-loss"
+	BlackHolePortFaultType    = "network-blackhole-port"
+	LatencyFaultType          = "network-latency"
+	PacketLossFaultType       = "network-packet-loss"
+	missingRequiredFieldError = "required parameter %s is missing"
 )
+
+type NetworkFaultRequest interface {
+	ValidateRequest() error
+	ToString() (string, error)
+}
+
+type NetworkBlackholePortRequest struct {
+	Port        *int    `json:"Port"`
```

Comment:
nit: a port is a 16-bit unsigned integer. Should we use `uint16` type for `Port`? That will rule out it ever being  negative or more than 65535.

Reply:
k